src/gui/attributetable/qgsfeaturelistview.cpp

```diff
@@ -338,3 +345,14 @@ void QgsFeatureListView::selectRow( const QModelIndex& index, bool anchor )
 
   mFeatureSelectionModel->selectFeatures( QItemSelection( tl, br ), command );
 }
+
+void QgsFeatureListView::setFeatureSelectionManager( QgsIFeatureSelectionManager* featureSelectionManager )
+{
+  if ( mFeatureSelectionManager )
+    delete mFeatureSelectionManager;
```

Comment:
You're completely right, I've just copied verbatim from https://github.com/qgis/QGIS/blob/master/src/gui/attributetable/qgsattributetableview.cpp#L124